tools/bbmap.py

```diff
@@ -28,38 +31,223 @@ def __init__(self, install_methods=None):
+                                    dedupe=True,
+                                    subs=subs,
+                                    passes=passes,
+                                    dupedist=dupedist,
+                                    k=kmer_size,
+                                    optical=optical,
+                                    spany=spany,
+                                    spanx=spanx,
+                                    adjacent=adjacent,
+                                    usetmpdir=True,
+                                    tmpdir=t_dir,
+                                    # if reads should be treated as unpaired, both 'unpair','repair' should be set to True
+                                    unpair=treat_as_unpaired,
+                                    repair=treat_as_unpaired,
+                                    containment=containment,
+                                    JVMmemory=JVMmemory,
+                                    **kwargs)
+                else:
+                    self.execute(tool='clumpify.sh', 
+                                    in1=inFastq1, in2=inFastq2,
+                                    out=outFastq1, out2=outFastq2,
+                                    dedupe=True,
+                                    subs=subs,
+                                    passes=passes,
+                                    dupedist=dupedist,
+                                    k=kmer_size,
+                                    optical=optical,
+                                    spany=spany,
+                                    spanx=spanx,
+                                    adjacent=adjacent,
+                                    usetmpdir=True,
+                                    tmpdir=t_dir,
+                                    # if reads should be treated as unpaired, both 'unpair','repair' should be set to True
+                                    unpair=treat_as_unpaired,
+                                    repair=treat_as_unpaired,
+                                    containment=containment,
+                                    JVMmemory=JVMmemory,
+                                    **kwargs)
+                # Make a list of reads to keep
+                with open(per_lb_read_list, 'at') as outf:
+                    for fq in (outFastq1, outFastq2):
+                        with util.file.open_or_gzopen(fq, 'rt') as inf:
+                            line_num = 0
+                            for line in inf:
+                                if (line_num % 4) == 0:
+                                    idVal = line.rstrip('\n')[1:]
+                                    if idVal.endswith('/1'):
+                                        outf.write(idVal[:-2] + '\n')
+                                line_num += 1
+
+                per_lb_read_lists.append(per_lb_read_list)
+        # merge per-library read lists together
+        util.file.concat(per_lb_read_lists, readListAll)
```

Comment:
Can't the read lists be written to a single file to start with instead of concat'ing later?

Reply:
The read lists are separate in prep for parallelizing across libraries.